pkg/pillar/types/dpc.go

```diff
@@ -660,43 +660,43 @@ func (config *DevicePortConfig) IsAnyPortInPciBack(
 // a corresponding Status struct.
 // Note that if fields are added the MostlyEqual function needs to be updated.
 type NetworkPortConfig struct {
-	IfName       string
-	USBAddr      string
-	PCIAddr      string
-	Phylabel     string // Physical name set by controller/model
-	Logicallabel string // SystemAdapter's name which is logical label in phyio
+	IfName       string `json:",omitempty"`
+	USBAddr      string `json:",omitempty"`
+	PCIAddr      string `json:",omitempty"`
+	Phylabel     string `json:",omitempty"` // Physical name set by controller/model
+	Logicallabel string `json:",omitempty"` // SystemAdapter's name which is logical label in phyio
 	// Unlike the logicallabel, which is defined in the device model and unique
 	// for each port, these user-configurable "shared" labels are potentially
 	// assigned to multiple ports so that they can be used all together with
 	// some config object (e.g. multiple ports assigned to NI).
 	// Some special shared labels, such as "uplink" or "freeuplink", are assigned
 	// to particular ports automatically.
-	SharedLabels []string
-	Alias        string // From SystemAdapter's alias
+	SharedLabels []string `json:",omitempty"`
+	Alias        string   `json:",omitempty"` // From SystemAdapter's alias
 	// Allow the local operator to make (limited) configuration changes to this
 	// network adapter using Local Profile Server.
-	AllowLocalModifications bool
+	AllowLocalModifications bool `json:",omitempty"`
 	// NetworkUUID - UUID of the Network Object configured for the port.
-	NetworkUUID uuid.UUID
-	IsMgmt      bool // Used to talk to controller
-	IsL3Port    bool // True if port is applicable to operate on the network layer
+	NetworkUUID uuid.UUID `json:",omitempty"`
+	IsMgmt      bool      `json:",omitempty"` // Used to talk to controller
+	IsL3Port    bool      `json:",omitempty"` // True if port is applicable to operate on the network layer
 	// InvalidConfig is used to flag port config which failed parsing or (static) validation
 	// checks, such as: malformed IP address, undefined required field, IP address not inside
 	// the subnet, etc.
-	InvalidConfig bool
-	Cost          uint8 // Zero is free
-	MTU           uint16
-	DhcpConfig
-	ProxyConfig
-	L2LinkConfig
-	WirelessCfg WirelessConfig
+	InvalidConfig bool   `json:",omitempty"`
+	Cost          uint8  `json:",omitempty"` // Zero is free
+	MTU           uint16 `json:",omitempty"`
+	DhcpConfig    `json:",omitempty"`
+	ProxyConfig   `json:",omitempty"`
+	L2LinkConfig  `json:",omitempty"`
+	WirelessCfg   WirelessConfig `json:",omitempty"`
```

Comment:
Does it work also for struct?
I read that it might not unless it is a pointer: https://stackoverflow.com/questions/33447334/golang-json-marshal-how-to-omit-empty-nested-struct

Reply:
As far as I understand the struct with "omitempty" will be omitted if all the field of this struct is empty. So, it kind of makes sense.
Nevertheless it looks like adding "ommitempty" on embedded structs are useless. Will fix.